### PR TITLE
Component Metadata Tracking Docs

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I added several sections to the docs that show how to use the component metadata introduced in #1151.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/advanced-guide/pipelines/materializers.md [hunk 1]
```diff
@@ -142,8 +142,8 @@ that are displayed in a dedicated way in the dashboard.
 See [src.zenml.metadata.metadata_types.MetadataType](https://github.com/zenml-io/zenml/blob/main/src/zenml/metadata/metadata_types.py)
 for more details.
 
-By default, this method will extract an artifact's storage size and runtime 
-data type, but you can overwrite it to track anything you wish. E.g., the 
+By default, this method will only extract the storage size of an artifact, but
+but you can overwrite it to track anything you wish. E.g., the
```
Comment (fa9r) [inline docs/book/advanced-guide/pipelines/materializers.md:146]:
```suggestion
By default, this method will only extract the storage size of an artifact, but
you can overwrite it to track anything you wish. E.g., the 
```

**Review by htahir1 — APPROVED** [review]

File: docs/book/component-gallery/orchestrators/orchestrators.md [hunk 2]
```diff
@@ -60,6 +60,19 @@ using the orchestrator is as simple as executing a python file which
 python file_that_runs_a_zenml_pipeline.py
 ```
 
+### Inspecting Runs in the Orchestrator UI
+
+If your orchestrator comes with a separate user interface (currently: Kubeflow,
```
Comment (strickvl) [inline docs/book/component-gallery/orchestrators/orchestrators.md:65]:
```suggestion
If your orchestrator comes with a separate user interface (for example: Kubeflow, 
```

Removed the currently as then this is something we have to maintain + remember to update.

File: docs/book/starter-guide/pipelines/fetching-pipelines.md [hunk 3]
```diff
@@ -141,30 +141,55 @@ zenml pipeline runs list -p <MY_PIPELINE_NAME_OR_ID>
 Each run has a collection of useful metadata which you can access to ensure all runs 
 are reproducible:
 
-#### git_sha
+#### Git SHA
 The Git commit SHA that the pipeline run was performed on. This will only be set
```
Comment (strickvl) [inline docs/book/starter-guide/pipelines/fetching-pipelines.md:145]:
```suggestion
The [Git commit SHA](https://www.mikestreety.co.uk/blog/the-git-commit-hash/) that the pipeline run was performed on. This will only be set 
```

File: docs/book/starter-guide/pipelines/fetching-pipelines.md [hunk 4]
```diff
@@ -141,30 +141,55 @@ zenml pipeline runs list -p <MY_PIPELINE_NAME_OR_ID>
...
 The status of a pipeline run can also be found here. There are four 
 possible states: failed, completed, running, cached:
 ```python
 status = run.status
 ```
 
-#### pipeline_configuration
+#### Configuration
 
 The `pipeline_configuration` is a super object that contains all configuration of
```
Comment (strickvl) [inline docs/book/starter-guide/pipelines/fetching-pipelines.md:161]: What is a 'super object'?

File: docs/book/starter-guide/pipelines/fetching-pipelines.md [hunk 5]
```diff
@@ -253,6 +280,19 @@ def some_step() -> Output(output_name=int):
 ```
 {% endhint %}
 
+### Output Artifact Metadata
+
+All output artifacts saved through ZenML will automatically have certain
+datatype-specific metadata saved with them. NumPy Arrays, for instance, always
+have their storage size, `shape`, `dtype`, and some statistical properties
```
Comment (strickvl) [inline docs/book/starter-guide/pipelines/fetching-pipelines.md:287]: Do we have a reference to API docs or just the code where we list exactly what properties are saved with them? Or are they embedded in each integration separately? Maybe give an example for where someone would go to find these properties for the NumPy arrays that you specify, just so people know where to look to find this out, aside from just running the code.

**Review by strickvl — CHANGES_REQUESTED** [review]

File: docs/book/starter-guide/pipelines/fetching-pipelines.md (hunk 5 above)
Comment (fa9r) [reply docs/book/starter-guide/pipelines/fetching-pipelines.md:287]: Yeah the contents of this dict change depending on the artifact. I guess you could look into the code of the respective materializer to figure it out but not sure that's what we want to lead users to do. IMO programmatically getting the metadata and just seeing what keys are in there is the best way to find out at the moment. Also, these properties are not designed as something that users will ever need to manually fetch since they will soon automatically be displayed in the dashboard next to the info of each run / step. I guess including screenshots of that in the docs will make it clearer once we have this.

File: docs/book/starter-guide/pipelines/fetching-pipelines.md (hunk 4 above)
Comment (fa9r) [reply docs/book/starter-guide/pipelines/fetching-pipelines.md:161]: An object that is really really nice and great 👍

Comment (htahir1) [reply docs/book/starter-guide/pipelines/fetching-pipelines.md:161]: HAHA i would rephrase this as `is an object`
